netscaler/netscaler_resource.go

```diff
@@ -123,16 +156,37 @@ func (c *NitroClient) doHTTPRequest(method string, url string, bytes *bytes.Buff
 	if err != nil {
 		return []byte{}, err
 	}
+	// For login request 'sessionid' is present in cookies
+	for _, cookie := range resp.Cookies() {
+		if cookie.Name == "sessionid" {
+			sessionid, err := url.QueryUnescape(cookie.Value)
+			if err == nil {
+				c.sessionid = sessionid
+			}
+		}
+	}
 	log.Println("[DEBUG] go-nitro: response Status:", resp.Status)
-	return respHandler(resp)
+	body, err := respHandler(resp)
+	// Clear sessionid in case of session-expiry
+	if resp.Status == "401 Unauthorized" {
+		var data map[string]interface{}
+		err2 := json.Unmarshal(body, &data)
+		if err2 == nil {
+			data["errorcode"] = int(data["errorcode"].(float64))
+			if data["errorcode"] == nsErrSessionExpired || data["errorcode"] == nsErrAuthTimeout {
+				c.sessionid = ""
+			}
+		}
+	}
+	return body, err
 }
 
 func (c *NitroClient) createResource(resourceType string, resourceJSON []byte) ([]byte, error) {
 	log.Println("[DEBUG] go-nitro: Creating resource of type ", resourceType)
 
 	url := c.url + resourceType
 
-	if !strings.HasSuffix(resourceType, "_binding") {
+	if !strings.HasSuffix(resourceType, "_binding") && !contains(idempotentInvalidResources, resourceType) {
```

Comment:
I am ok with how you determine when to omit the `idempotent=true` query parameter.

In the long term it places the burden on go-nitro to maintain this list.

As a note I think this should be exposed to the User in an extensible manner much like python does`**kwargs`
The only way for this to happen would be to create some options struct and add there the options we need the user to define.
```
type NitroClientOptions struct {
Idempotent bool,
}
```
Then we would have to expose this option to all the methods calling this one. Essentially `NitroClient.AddResource` `NitroClient.DeleteResource` etc.

To avoid API incompatibility we could introduce variants of the default actions e.g. `NItroClient.AddResourceWithopts`

As I said it is not required for you to do this just putting it here for consideration.
@chiradeep  what do you think?

Reply:
Agreed with this approach @giorgos-nikolopoulos @imkrithesh @dheerajng

Reply:
Actually, I changed my mind. I prefer the way it is done in this PR. But a way to "escape" or special case this when the SDK has not kept pace with the server-side release would be nice to have

Reply:
I would also prefer to have the idempotent changes as a separate PR